### PR TITLE
[Issue][Frontend] Refactor the method of fetching data using @tanstack/react-query

### PR DESCRIPTION
**Pull request by badukwei**
Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary

Refactor the method of fetching data using @tanstack/react-query

## Linked Issue

close https://github.com/social-tw/social-tw-website/issues/326

## Details

Modify fetching singlePost data and comments

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities
-   [x] Run `yarn lint:fix`

### CODE REVIEW

File: packages/frontend/src/hooks/useFetchSinglePost.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        epochKey: fetchedPost.epochKey,
+        content: fetchedPost.content,
+        publishedAt: new Date(Number(fetchedPost.publishedAt)),
+        commentCount: fetchedPost.commentCount,
+        upCount: fetchedPost.upCount,
+        downCount: fetchedPost.downCount,
+        isMine: isMine,
+        finalAction: finalAction,
+        status: PostStatus.Success,
+    }
```
Comment (Xiawpohr) [inline packages/frontend/src/hooks/useFetchSinglePost.ts:49]: L29 - L49 可以用 useMemo 包起來，讓 post 可以因 userState 改變而更新。例如：
```js
const post = useMemo(() => {
    if (!fetchedPost) return

    let isMine = false
    let finalAction = null
    if (userState) {
        const voteCheck = checkVoteIsMine(fetchedPost.votes, userState)
        isMine = voteCheck.isMine
        finalAction = voteCheck.finalAction
    }
  
    return {
          id: fetchedPost._id,
          postId: fetchedPost.postId,
          epochKey: fetchedPost.epochKey,
          content: fetchedPost.content,
          publishedAt: new Date(Number(fetchedPost.publishedAt)),
          commentCount: fetchedPost.commentCount,
          upCount: fetchedPost.upCount,
          downCount: fetchedPost.downCount,
          isMine: isMine,
          finalAction: finalAction,
          status: PostStatus.Success,
    }
}, [fetchedPost, userState])
```

File: packages/frontend/src/pages/PostDetail.tsx [hunk 2]
```diff
@@ -66,35 +68,35 @@ export default function PostDetail() {
...
+    //             upCount: post.upCount,
+    //             downCount: post.downCount,
+    //             isMine: isMine,
+    //             finalAction: finalAction,
+    //             status: PostStatus.Success,
+    //         })
+    //     }
+
+    //     loadPost()
+    // }, [id, userState])
```
Comment (Xiawpohr) [inline packages/frontend/src/pages/PostDetail.tsx:99]: 如果這段程式碼沒用的話，就可以刪除了。如果還有其他考量的話，請再留下一段註解說明為什麼這段程式碼被註解了

Comment (badukwei) [reply packages/frontend/src/pages/PostDetail.tsx]: 已刪除

**Review by Xiawpohr — APPROVED** [review]
Approved this pull request

**Pull request merged**